dockerplugin.py

```diff
@@ -229,6 +248,62 @@ def read_memory_stats(container, dimensions, stats, t):
                    .format(_c(container)))
 
 
+def read_cpu_shares_stats(container,
+                          container_inspect, cstats,
+                          cpu_percent, sum_of_shares):
+    # Get cpu shares used by container
+    stats = cstats.stats
+    dimensions = cstats.dimensions
+    num_cpus_host = len(stats['cpu_stats']['cpu_usage']['percpu_usage'])
+    shares_used_percent = 0.0
+    cpu_shares = container_inspect['HostConfig']['CpuShares'] or \
+        DEFAULT_SHARES
+    fraction_of_shares = cpu_shares / float(sum_of_shares)
+    shares_used_percent = cpu_percent / num_cpus_host / fraction_of_shares
+    emit(container, dimensions,
+         'cpu.shares',
+         [shares_used_percent],
+         type_instance='used.percent',
+         t=stats['read'])
+
+
+def read_cpu_quota_stats(container, container_inspect, cstats):
+    stats = cstats.stats
+    dimensions = cstats.dimensions
+    host_config = container_inspect['HostConfig']
+    cpu_quota = host_config.get('CpuQuota', 0)
+
+    if not cpu_quota:
+        return
+
+    if 'preread' in stats and 'precpu_stats' in stats:
+        period = host_config.get('CpuPeriod', 0)
```

Comment:
Minor: why not make `100000` the default the assign to `cpu_period`?

Reply:
If the value is not set, docker engine returns a 0 value for cpu period by default. But the actual default value is 100000. Hence the additional check.